### ISSUE TITLE
dependency on MCMCpack seems unnecessary

### ISSUE THREAD

**Issue by fabian-s** [refunders/refund#57]
MCMCpack is a shitty dependency to have because it needs two Bioconductor packages: graph and Rgraphviz. 

So for people to install refund without warnings and errors, they'd need to know to install these two from Bioconductor.

None of the other dependencies need any Bioconductor stuff, I think.

AFAICS we depend on MCMCpack only because of the `riwish`-RNG, used in [Gibbs_CS_Wish](https://github.com/refunders/refund/blob/c14beb5cccf49236f71762f539dfd1dc12fec9f2/R/Gibbs_CS_Wish.R#L161):

```
inv.sig = solve(riwish(v + I, Psi + t(resid.cur) %*% resid.cur))
```

and [gibbs_mult_wish](https://github.com/refunders/refund/blob/c14beb5cccf49236f71762f539dfd1dc12fec9f2/R/Gibbs_Mult_Wish.R#L208):

```
inv.sig = solve(riwish(v + IJ, Psi + t(resid.cur) %*% resid.cur))
```

This draws an "inverse Wishart" RV that immediately gets inverted, so a Wishart RV. 

@jeff-goldsmith: Couldn't we just use `stats::rWishart(...)` instead of `solve(riwishart(...))`, save some cycles by not doing the inversion (in each iteration, no less!), and drop the dependency on MCMCpack?

**Comment by jeff-goldsmith** [comment]
should be doable, yeah. the offending code also shows up in Gibbs_Mult_Wish, but the fix will be the same there. i’ll make an update and do some testing to make sure everything works post-enar, if that timeline suffices.

On Mar 2, 2016, at 10:40 AM, Fabian Scheipl <notifications@github.com<mailto:notifications@github.com>> wrote:

MCMCpack is a shitty dependence to have because it needs two Bioconductor packages: graph and Rgraphviz.

So for people to install refund without warnings and errors, they'd need to know to install these two from Bioconductor.

None of the other dependencies need any Bioconductor stuff, I think.

AFAICS we depend on MCMCpack only because of the riwish-RNG, used in a single line in Gibbs_CS_Wishhttps://github.com/refunders/refund/blob/c14beb5cccf49236f71762f539dfd1dc12fec9f2/R/Gibbs_CS_Wish.R#L161:

inv.sig = solve(riwish(v + I, Psi + t(resid.cur) %*% resid.cur))

This draws an "inverse Wishart" RV that immediately gets inverted, so a Wishart RV.

@jeff-goldsmithhttps://github.com/jeff-goldsmith: Couldn't we just use stats::rWishart(...) instead of solve(riwishart(...)), save some cycles by not doing the inversion (in each iteration, no less!), and drop the dependency on MCMCpack?

**Comment by fabian-s** [comment]
cool, thanks!

**Comment by jeff-goldsmith** [comment]
should be fixed by 8e47167 -- sorry for taking so long!

**Comment by fabian-s** [comment]
Thanks @jeff-goldsmith !
